frontend/src/components/pages/NewAccountPage.tsx

```diff
@@ -39,6 +63,33 @@ const NewAccountPage = (): React.ReactElement => {
     { loading: createVolunteerLoading, error: createVolunteerError },
   ] = useMutation(CREATE_VOLUNTEER_USER);
 
+  const [deleteUserInvite, { error: deleteUserInviteError }] = useMutation(
+    DELETE_USER_INVITE,
+  );
+
+  const [userInvite, setUserInvite] = useState<UserInviteResponseDTO>();
+
+  const history = useHistory();
+
+  const queryParams = new URLSearchParams(window.location.search);
+  const token = queryParams.get("token");
+
+  useQuery<GetUserInviteResponse>(GET_USER_INVITE, {
+    fetchPolicy: "cache-and-network",
+    variables: {
+      uuid: token,
+    },
+    onCompleted: (data) => {
+      setUserInvite(data.getUserInvite);
+      if (userInvite?.role === "EMPLOYEE") {
+        setIsAdmin(true);
+      }
```

Comment:
This part is buggy from testing. Should not directly use state after using set action as it is not synchronous.
```suggestion
      if (data.getUserInvite?.role === "EMPLOYEE") {
        setIsAdmin(true);
      }
```

Reply:
I think after this gets changed, this PR should be good to change.